docs/DeveloperGuide.md

```diff
@@ -8,6 +8,139 @@
+
+__Implementation Details__
+
+*Class Diagram*: Illustrates `AppUi` with its methods for displaying messages and capturing user input.
+
+{insert class diagram for UI Component here}
+
+__Methods__
+
+- **displayWelcomeMessage()**: Outputs a startup message.
+- **getUserInput()**: Reads input from the user.
+- **showUnknownCommandMessage()**: Notifies the user of an unrecognized command.
+- **showErrorMessage(String message)**: Displays a specific error message.
+
+__Usage Example__
+
+```
+AppUi ui = new AppUi();
+ui.displayWelcomeMessage();
+String userInput = ui.getUserInput();
+ui.showUnknownCommandMessage();
+```
+
+__Design Considerations__
+
+- **Future Extension**: Additional Ui features such as graphical interfaces or web-based interactions may be added to enhance user experience.
+
+### Parser Component
+__Overview__
+
+The Parser component includes `InputParser` and `DateParser`. `InputParser` processes user commands, while `DateParser` validates date string.
+
+__Class Structure__
+
+- **Attributes**:
+  - `formatter`: `DateTimeFormatter` — Defines a date format for parsing.
+
+__Implementation Details__
+
+*Class Diagram*: Shows `InputParser` parsing command input and `DateParser` handling date validation.
+
+{insert class diagram for Parser Component here}
+
+__Methods__
+
+- **InputParser.parseCommands(String input)**: Breaks down commands and arguments.
+- **DateParser.parse(String dateStr)**: Validates and converts date strings.
+
+__Usage Example__
+
+```
+HashMap<String, String> commandArgs = InputParser.parseCommands("add /date 12/10/24 /amount 500");
+LocalDate parsedDate = DateParser.parse("12/10/24");
+```
+
+__Design Considerations__
+
+- **Future Extension**: To support more complex commands and argument parsing, the Parser component could introduce additional parsers, such as `CommandParser` and `ArgumentParser`, extending from an abstract base. Supporting alternative date formats in `DateParser` could enhance flexibility, accommodating user input from different locales or formats.
+
+---
+### Logic
+__Overview__
+
+The Logic component manages core functionalities in the application like adding, editing and deleting financial entries. 
+It interacts with `FinancialList`, `AppUi` and `Storage`, and leverages command classes (`AddExpenseCommand`, `AddIncomeCommand`, etc.) to execute operations.
+
+__Class Structure__
+
+- **Attributes**:
+  - `financialList`: `FinancialList` — Stores financial entries.
+  - `ui`: `AppUi` - Manages user interactions.
+  - `storage`: `Storage` - Handles data persistence
+
+__Implementation Details__
+
+*Class Diagram*: Shows Logic as the main controller interacting with `FinancialList`, `AppUi`, and various command classes for operations (e.g., `AddIncomeCommand`, `DeleteCommand`).
+
+{Insert Class Diagram for Logic Component here}
+
+__Constructor__
+
+The Logic constructor initializes `FinancialList`, `AppUi`, and `Storage` components to support all operations.
+
+__Methods__
```

Comment:
I am pretty sure this is not quite right. Could you amend this part to accurately reflect the key methods of the class?